src/models/mmt_encoder.py

```diff
@@ -0,0 +1,202 @@
+     elif relative_vocab_size < (
+         etc_feature_utils.RelativePositionGenerator(
+             relative_pos_max_distance).relative_vocab_size +
+             _NUM_OTHER_RELATIVE_IDS):
+       raise ValueError(
+           f'`relative_vocab_size` ({relative_vocab_size}) too small for '
+           f'`relative_pos_max_distance` ({relative_pos_max_distance}')
+
+     if embedding_size is None:
+       embedding_size = hidden_size
+
+     activation = tf.keras.activations.get(inner_activation)
+     initializer = tf.keras.initializers.TruncatedNormal(
+         stddev=initializer_range)
+     initializer = tf.keras.initializers.get(initializer)
+
+     self._word_embedding_layer = etc_layers.EmbeddingLookup(
+         vocab_size=vocab_size,
+         embedding_size=embedding_size,
+         projection_size=hidden_size,
+         initializer_range=initializer_range,
+         name='word_embeddings')
+
+     if max_absolute_position_embeddings is not None:
+       self._position_embedding_layer = layers.PositionEmbedding(
+           initializer=initializer,
+           max_length=max_absolute_position_embeddings,
+           name='absolute_position_embeddings')
+     else:
+       self._position_embedding_layer = None
+
+     self._segment_embedding_layer = etc_layers.EmbeddingLookup(
+         vocab_size=segment_vocab_size,
+         embedding_size=embedding_size,
+         projection_size=hidden_size,
+         initializer_range=initializer_range,
+         use_one_hot_lookup=use_one_hot_lookup,
+         name='segment_embeddings')
+     
+     self._patch_embedding_projection = tf.keras.layers.Dense(
+         units=hidden_size,
+         kernel_initializer=initializer,
+         name='patch_embedding_projection')
+
+     self._embedding_norm_layer = tf.keras.layers.LayerNormalization(
+         axis=-1, epsilon=1e-12, name='embeddings/layer_norm')
+
+     self._embedding_dropout_layer = tf.keras.layers.Dropout(
+         rate=hidden_dropout_prob, name='embeddings/dropout')
+
+     self._relative_transformer = etc_layers.RelativeTransformerLayers(
+         hidden_size=hidden_size,
+         num_hidden_layers=num_hidden_layers,
+         num_attention_heads=num_attention_heads,
+         intermediate_size=intermediate_size,
+         hidden_act=activation,
+         hidden_dropout_prob=hidden_dropout_prob,
+         attention_probs_dropout_prob=attention_probs_dropout_prob,
+         initializer_range=initializer_range,
+         relative_vocab_size=relative_vocab_size,
+         use_pre_activation_order=use_pre_activation_order,
+         use_one_hot_lookup=use_one_hot_lookup)
+     
+  def call(self,
+           word_ids: tf.Tensor,
+           segment_ids: Optional[tf.Tensor] = None,
+           att_mask: Optional[tf.Tensor] = None,
+           relative_att_ids: Optional[tf.Tensor] = None,
+           patch_embeddings: Optional[tf.Tensor] = None,
+           training: Optional[bool] = None):
+
+    """Calls the layer.
+      Args:
```

Comment:
nit: some times you are leaving blank space lines in-between the description, the "args" and the "returns" section, and some times you are not. Please uniformize.